swarm/api/http/server_test.go

```diff
@@ -892,3 +894,239 @@ func TestMethodsNotAllowed(t *testing.T) {
 	}
 
 }
+
+// HTTP convenience function
+func httpDo(httpMethod string, url string, reqBody io.Reader, headers map[string]string, verbose bool, t *testing.T) (*http.Response, string) {
+	// Build the Request
+	req, _ := http.NewRequest(httpMethod, url, reqBody)
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+	if verbose {
+		t.Log(req.Method, req.URL, req.Header, req.Body)
+	}
+
+	// Send Request out
+	httpClient := &http.Client{}
+	res, err := httpClient.Do(req)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Read the HTTP Body
+	buffer, _ := ioutil.ReadAll(res.Body)
+	defer res.Body.Close()
+	body := string(buffer)
+
+	return res, body
+}
+
+func TestGet(t *testing.T) {
+	verbose := false
+	// Setup Swarm and upload a test file to it
+	srv := testutil.NewTestSwarmServer(t, serverFunc)
+	defer srv.Close()
+
+	// Accept: text/html GET / -> 200 HTML, Swarm Landing Page
+	getRootHTML := func() {
+		headers := make(map[string]string)
+		headers["Accept"] = "text/html"
+		res, body := httpDo("GET", fmt.Sprintf("%s/", srv.URL), nil, headers, verbose, t)
+
+		if res.StatusCode != 200 {
+			t.Fatal("expected GET / to return a 200 but it didn't")
+		}
+		if !strings.HasPrefix(body, "<html>") {
```

Comment:
i would match a against a text we know precisely will show up (i.e. something inside the `<body>` tag like you did in the tests below

Reply:
done